src/Saturn.AzureFunctions/AzureFunctions.fs

```diff
@@ -5,39 +5,55 @@ open Giraffe
 open System
 open Microsoft.AspNetCore.Http
 open System.Threading.Tasks
-open Microsoft.Azure.WebJobs.Host
+open Microsoft.Extensions.Logging
+open Giraffe.Serialization.Json
+open Giraffe.Serialization.Xml
 
 module AzureFunctions =
 
   type FunctionState = {
-    Logger: TraceWriter option
+    Logger: ILogger option
     Router: HttpHandler option
     ErrorHandler: (System.Exception -> HttpHandler)
     NotFoundHandler: HttpHandler
     HostPrefix: string
+    JsonSerializer: IJsonSerializer
+    XmlSerializer: IXmlSerializer
+    NegotiationConfig: INegotiationConfig
   }
 
   type FunctionBuilder internal () =
-    member val LogWriter : TraceWriter option = None with get,set
+    member val LogWriter : ILogger option = None with get,set
 
     member __.Yield(_) =
         let errorHandler (ex : Exception) =
-            __.LogWriter |> Option.iter (fun logger -> logger.Error("An unhandled exception has occurred while executing the request.", ex))
+            __.LogWriter |> Option.iter (fun logger -> logger.LogError("An unhandled exception has occurred while executing the request.", ex))
             clearResponse >=> Giraffe.HttpStatusCodeHandlers.ServerErrors.INTERNAL_ERROR ex.Message
 
         let notFoundHandler =
             clearResponse >=> Giraffe.HttpStatusCodeHandlers.RequestErrors.NOT_FOUND "Not found"
-        {Logger = None; Router = None; ErrorHandler = errorHandler; NotFoundHandler = notFoundHandler; HostPrefix = "/api"}
+        {Logger = None; Router = None; ErrorHandler = errorHandler; NotFoundHandler = notFoundHandler; HostPrefix = "/api"; JsonSerializer = NewtonsoftJsonSerializer(NewtonsoftJsonSerializer.DefaultSettings); XmlSerializer = DefaultXmlSerializer(DefaultXmlSerializer.DefaultSettings);  NegotiationConfig = DefaultNegotiationConfig() }
 
     member __.Run(state: FunctionState) : (HttpRequest -> Task<HttpResponse>) =
+      let wrapGiraffeServices (existing:IServiceProvider) =
+        { new IServiceProvider with
+            member __.GetService(t:Type) =
+              if (t = typeof<IJsonSerializer>) then upcast state.JsonSerializer
+              elif (t = typeof<IXmlSerializer>) then upcast state.XmlSerializer
+              elif (t = typeof<INegotiationConfig>) then upcast state.NegotiationConfig
+              else existing.GetService(t)
+        }
+
+
       let next : HttpContext -> Task<HttpContext option> = Some >> Task.FromResult
       fun req ->
+        req.HttpContext.RequestServices <- wrapGiraffeServices req.HttpContext.RequestServices
         let r =
           match state.Router with
           | Some r -> r
           | None -> failwith "Router needs to be defined"
         let r = router {
-          pipe_through (fun nxt ctx -> state.Logger |> Option.iter (fun log -> ctx.Items.["TraceWriter"] <- log); nxt ctx)
+          pipe_through (fun nxt ctx -> state.Logger |> Option.iter (fun log -> ctx.Items.["Logger"] <- log); nxt ctx)
```

Comment:
Hmm, I wonder if this module should also provide a helper or an extension on HttpContext to get the Logger being set here vs relying on people knowing and likely making their own helper. What do you think?